crates/solidity/outputs/cargo/tests/src/doc_examples/using_the_cursor.rs

```diff
@@ -88,36 +88,31 @@ fn using_the_cursor() -> Result<()> {
 
     {
         // --8<-- [start:using-iterator-api]
-        let cursor = parse_output.create_tree_cursor();
-
-        let identifiers: Vec<_> = cursor
-            .filter_map(|node| {
-                node.as_terminal_with_kind(TerminalKind::Identifier)
-                    .cloned()
-            })
-            .map(|identifier| identifier.text.clone())
+
+        let identifiers: Vec<_> = parse_output
+            .tree()
+            .descendants()
+            .filter(|node| node.is_terminal_with_kind(TerminalKind::Identifier))
+            .map(|identifier| identifier.unparse())
             .collect();
 
         assert_eq!(identifiers, &["Foo", "Bar", "Baz"]);
         // --8<-- [end:using-iterator-api]
     }
 
     {
-        // --8<-- [start:using-labeled-cursors]
-        let cursor = parse_output.create_tree_cursor();
+        // --8<-- [start:using-cursors-with-labels]
 
-        let identifiers: Vec<_> = cursor
-            .with_edges()
+        let identifiers: Vec<_> = parse_output
+            .tree()
+            .descendants()
             .filter(|node| node.label == Some(EdgeLabel::Name))
-            .filter_map(|node| {
-                node.as_terminal_with_kind(TerminalKind::Identifier)
-                    .cloned()
-            })
-            .map(|identifier| identifier.text.clone())
+            .filter(|node| node.is_terminal_with_kind(TerminalKind::Identifier))
```

Comment:
It's weird that `node` here is actually an `Edge`, which in turn consists of a `label` and a `node`. I think that's the most confusing part of this change in general: the items of both `children()` and `CursorIterator` (through `descendants()`) are now edges instead of nodes. Previously it was explicit with the added `.with_edges()` call to the cursor. It'll maybe catch users by surprise.

Reply:
We only had `.nodes()` and `.edges()` because edges came later, but I'm not sure it makes sense to add a specific iterator for it, given that it doesn't cost anything to do both, and they already have a `Deref` between them..
I updated the variable names to make this clearer. Please let me know if you have any other suggestions.